OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -224,7 +228,7 @@ internal static void ProcessOrder(OrderManager orderManager, World world, int cl
 							Client = info,
 							Mod = mod.Id,
 							Version = mod.Metadata.Version,
-							Password = orderManager.Password,
+							Password = CryptoUtil.SHA1Hash(request.AuthToken + orderManager.Password),
```

Comment:
SHA1 shouldn't be used for security stuff anymore, use either SHA2 or SHA3.

Reply:
This isn't trying to fix active MITM, it is aiming to fix the password being saved in plaintext in replays.

Reply:
Then what about simply not storing the handshake in the replay? It is useless anyway and not storing it has multiple other advantages over your current proposal.

Reply:
The replay code does not allow this.